dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java

```diff
@@ -83,48 +188,96 @@ private static ContentType createTestType(final ContentTypeAPI contentTypeAPI)
+     * - If the user have {@link PermissionLevel#CAN_ADD_CHILDREN} over the ContentType
+     * - If the user doesn't have any permission should throw a DotSecurityException when try to save it
+     * - If the user just has {@link PermissionLevel#CAN_ADD_CHILDREN} over the save {@link ContentType} then should throw a DotSecurityException when try to save it
+     */
     @Test
-    public void test_Publish_With_Save_Contentlet_Actionlet_Tag () throws DotSecurityException, DotDataException {
+    @UseDataProvider("usersAndContentTypeWithoutHostField")
+    public void test_Publish_With_Save_Contentlet (final TestCase testCase) throws DotSecurityException, DotDataException {
+        final WorkflowAPI workflowAPI = APILocator.getWorkflowAPI();
 
-        final Contentlet contentlet = new Contentlet();
-        contentlet.setContentType(customContentType);
-        contentlet.setProperty("title", "Test");
-        contentlet.setProperty("txt", "Test");
-        contentlet.setProperty("tag", "test");
+        final Contentlet contentlet = new ContentletDataGen(testCase.contentType.id())
+            .setProperty("title", "Test")
+            .setProperty("txt", "Test")
+            .setProperty("tag", "test")
+            .next();
 
-        final Contentlet contentletSaved =
-                workflowAPI.fireContentWorkflow(contentlet,
-                    new ContentletDependencies.Builder()
-                        .modUser(APILocator.systemUser())
-                        .workflowActionId(SystemWorkflowConstants.WORKFLOW_SAVE_ACTION_ID)
-                        .build());
+        Contentlet contentletSaved = null;
 
-        Assert.assertNotNull(contentletSaved);
-        Assert.assertEquals("Test", contentletSaved.getStringProperty("title"));
-        Assert.assertEquals("Test", contentletSaved.getStringProperty("txt"));
+        try {
+            contentletSaved =
+                    workflowAPI.fireContentWorkflow(contentlet,
+                            new ContentletDependencies.Builder()
+                                    .modUser(testCase.user)
+                                    .respectAnonymousPermissions(testCase.respectFrontendRoles)
+                                    .workflowActionId(SystemWorkflowConstants.WORKFLOW_SAVE_ACTION_ID)
+                                    .build());
 
-        final List<TagInode> tagInodes = APILocator.getTagAPI().getTagInodesByInode(contentletSaved.getInode());
-        Assert.assertNotNull(tagInodes);
-        Assert.assertFalse(tagInodes.isEmpty());
+            checkContentSaved(contentletSaved);
+
+            Assert.assertTrue(
+                    testCase.hasContentTypeAddChildrenPermission &&
+                            testCase.hasSaveActionPermission ||
+                            (testCase.respectFrontendRoles && this.haveFrontendPermission(testCase.contentType))
+            );
+        } catch(Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)